examples/resources/cloudflare_ruleset/resource.tf

```diff
@@ -383,3 +383,22 @@ resource "cloudflare_ruleset" "http_config_rules_example" {
     enabled     = true
   }
 }
+
+# Set compress algorithm for response.
+resource "cloudflare_ruleset" "response_compress_brotli_html" {
+  zone_id     = "0da42c8d2132a9ddaf714f9e7c920711"
+  name        = "Brotli response compression for HTML"
+  description = "Response compression ruleset"
+  kind        = "zone"
+  phase       = "http_response_compression"
+
+  rules {
+    action = "compress_response"
+    action_parameters {
+      algorithms = [ { name: "brotli" }, { name: "default" } ]
```

Comment:
this syntax isn't quite going to work given the current schema in Terraform. do we need the `name` property exist? if you're after an order dependent list, we could use `TypeList` like this: 

```hcl
action_parameters {
  algorithms = ["brotli", "default"]
}
```

however, if we _may_ add other properties down the line, we want to update this to be nested objects in the schema.

Reply:
> however, if we may add other properties down the line, we want to update this to be nested objects in the schema.

That's the idea.  We might, one day, add extra properties.

I've updated the test to follow the same pattern as `headers` (action parameter for the `rewrite` action).  Let me know if that's not the best approach.

Reply:
that's fine for now. terraform doesn't really have a nested set of nestable objects like you're trying to do here.